app/models/transaction.rb

```diff
@@ -0,0 +1,8 @@
+class Transaction < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :transaction_categories
+  has_many :categories, through: :transaction_categories
+
+  validates :name, presence: true, length: { maximum: 255 }
+  validates :amount, presence: true, numericality: { greater_than: 0, only_integer: true }
+end

```

Comment:
- [ ]  As per this [note](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=for%20your%20project-,(please%20note%3A%20%22Transaction%22%20is%20a%20name%20already%20used%20by%20ActiveRecord%2C%20so%20using%20it%20as%20a%20name%20for%20your%20model%20and%20table%20will%20result%20in%20an%20error,-).), "Transaction" is a name already used by ActiveRecord, so using it as a name for your model and table will result in an error Please change it (controller, model, and views). you can name it `payment` instead.

Reply:
Done! ✅